src/utils.ts

```diff
@@ -228,3 +234,30 @@ export function getUniq(arr: string[]): string[] {
 
     return uniq;
 }
+
+/**
+ * Check if the line is a block end.
+ * @param line The textual representation of the line.
+ * @return true if the specified line is a block end, false otherwise.
+ */
+export function isBlockEndLine(line: string) : boolean {
+    return /^ *#\+END(_|:)/i.test(line);
```

Comment:
Would be better to use `\s` rather than ` ` space. This would also work if someone have a tab indented source block